e2e/tests/text/text-editing.spec.ts

```diff
@@ -1,53 +1,89 @@
-import { test } from '@playwright/test';
-import {
-  addTextbox,
-  getObjectCenter,
-  clickCanvas,
-  getObjectControlPoint,
-} from '../../utils/objects';
-test('text typing itext vs textbox', async ({ page }) => {
+import { expect, test } from '@playwright/test';
+import { TestUtil } from '../../utils/TestUtil';
+import '../../utils/setupTest';
+
+test('textbox typing and resizing', async ({ page }) => {
+  const util = new TestUtil(page);
   await page.goto('http://127.0.0.1:8080/e2e/site');
-  const textboxID = 'text1';
-  await addTextbox(page, textboxID, 'initial text', {
+  const textboxUtil = await util.addTextbox('initial text', {
     width: 200,
     left: 50,
   });
-  const textCenter = await getObjectCenter(page, textboxID);
-  await clickCanvas(page, {
+  const textCenter = await textboxUtil.getObjectCenter();
+
+  expect(await util.screenshot()).toMatchSnapshot({ name: 'initial' });
+
+  await util.clickCanvas({
     position: textCenter,
     clickCount: 2,
     delay: 200,
   });
 
+  expect(await util.screenshot()).toMatchSnapshot({ name: 'start' });
+
   await page
     .locator('textarea')
     .type(
       'insert text in a textbox from the keyboard will wrap text on current textbox width',
       {
-        delay: 160,
+        // delay: 160,
       }
     );
 
-  const controlPoint = await getObjectControlPoint(page, textboxID, 'mr');
+  expect(await util.screenshot()).toMatchSnapshot({ name: 'typed' });
+
+  const mrControlPoint = await textboxUtil.getObjectControlPoint('mr');
 
   // click outside to stop editing
-  await clickCanvas(page, {
+  await util.clickCanvas({
     position: {
-      x: controlPoint.x + 20,
-      y: controlPoint.y,
+      x: mrControlPoint.x + 20,
+      y: mrControlPoint.y,
     },
     clickCount: 1,
     delay: 200,
   });
+  expect(await util.screenshot()).toMatchSnapshot({ name: 'exit_editing' });
 
   // click the object to select it
-  await clickCanvas(page, {
+  await util.clickCanvas({
     position: textCenter,
   });
 
-  // drag the control
-  await page.mouse.move(controlPoint.x, controlPoint.y);
+  // drag the mr control
+  await page.mouse.move(mrControlPoint.x, mrControlPoint.y);
```

Comment:
In our project we create in Cypress a utility to drag from-to in the canvas because that's the most common interaction almost